src/lib/fountain/FountainMetadata.h

```diff
@@ -65,9 +65,12 @@ class FountainMetadata
 		return res;
 	}
 
-	void increment_block_id()
+	void increment_block_id(unsigned radioactive_block_id)
 	{
-		update_block_id_internal(block_id()+1, _data.data()+4);
+		unsigned next = block_id()+1;
+		if (next == radioactive_block_id)
+			next += 1;
```

Comment:
The gist is: there might be a block_id we never use (due to its size not matching). If we have such a block_id, we pass it in and increment past it.